source/_posts/2018-03-14-GoogleCast.md

```diff
@@ -0,0 +1,401 @@
+---
+title:  GoogleCast
+date: 2018-03-14 10:00:00
+tags: ios,google,googlecast
+authorIds:
+- anho
+categories:
+- iOS
+---
+
+Today, on the AppStore, you are able to find a rapidly increasing number of apps that will allow their users to watch video content. There might come a time when you will have to develop such an app, and functionality like GoogleCast and AirPlay is not only nice to have, but also expected by the end user.
+
+In this post we will tackle together the steps required in order to allow your users cast videos to a GoogleCast device.
+
+## Let's get started
+In order to be able to solely focus on integrating the GoogleCast SDK to our app, I have created a demo project that you can use in order to follow along.
+
+The project containt:
+
+- Added GoogleCast SDK via CocoaPods [Setup Instructions](https://developers.google.com/cast/docs/ios_sender_setup#xcode_setup)
+- An AVPlayer that allows users to play video content locally
+
+As well you will need to register your application to Google Cast SDK Developer Console at: https://cast.google.com/publish/#/signup
+
+You can download the demo project here: [Started Project](https://github.com/nodes-projects/cast-nodes-ios/tree/cast-demo-start)
+
+
+### Part 1: Connecting to a GoogleCast device
+
+To establish a connection with a device we need our app to:
+
+- initialise the GCKCastContext
+- create a GCKSessionManager
+- add a the GCKUICastButton
+
+For us to be able to do in a nice way the things mentioned above we need to create a CastManager, a Singleton instance in our case.
+
+In our manager we now create  `func initialise()` where we will do the initial setup. Here we will call `func initialiseContext()` and `createSessionManager()`
+
+Go ahead declare your `sessionManager`, which will be in charge of handling the connection to a device, and create the functions as following:
+
+```
+private var sessionManager: GCKSessionManager!
+
+func initialise() {
+    initialiseContext()
+    createSessionManager()
+}
+
+//creates the GCKSessionManager
+private func createSessionManager() {
+    sessionManager = GCKCastContext.sharedInstance().sessionManager
+}
+
+//initialises the GCKCastContext
+private func initialiseContext() {
+    //application Id registered from the registered application
+    let options = GCKCastOptions(discoveryCriteria: GCKDiscoveryCriteria.init(applicationID: "EC926046"))
```

Comment:
should you change "EC92046" to <Your ID here> or something...are we exposing an ID that shouldn't be exposed?